src/main/scala/org/hatdex/hat/api/models/ApiData.scala

```diff
@@ -9,8 +9,23 @@
 
 package org.hatdex.hat.api.models
```

Comment:
`ApiData.scala` is not the best place for the new API models. The file refers to some old models that were used for some specific HAT Raw Data API models and have been deprecated since. Suggest moving it to a new file under a title of `RequestUtils` or something similar.

Reply:
As a general rule the word Utils is advised against (since everything can be a util) and Request is overloaded
I created a new class called query, i dont think works as well as Api but its the best i could come up with

> There are only two hard things in Computer Science: cache invalidation and naming things.

Reply:
Same here, suggested `RequestUtils` for the lack of better name that wouldn't conflict with existing classes. `Query` sounds misleading as the payload class is not directly related to the query parameters.
To get around that I then suggest to split it into two files: `ApplicationQuery` placed under `applications` package and separately `PaginatedResponse` placed separately.

Reply:
Yeah I thought Payload was ok in query since its a response wrapper but you are right i dont think it works

There are 4 classes in question:
* PaginationParameters (used for both input and output of pagination params)
* FilterParameters (input for filter params[Complex type])
* ApplicationFilters (output for filter params [String value of Complex type])(I wish this could be avoided)
* PayloadWrapper(data: JsValue,
                           next: Option[String] = None,
                           paginationParameters: PaginationParameters,
                           filters: ApplicationFilters)

Reply:
Open to suggestions

Reply:
Everything except the PayloadWrapper can go into `ApplicationQuery` class as all three classes are concerned with how specifically application queries are made.

Reply:
Ok my only thought it here is that since PayloadWrapper has a  dependency on  PaginationParameters and ApplicationFilters is it therefor also specific to Applicaiton